mos_tests/nova/nova_test.py

```diff
@@ -601,3 +606,80 @@ def test_boot_instance_from_volume_bigger_than_flavor_size(self):
         # Check that instance is reachable
         ping = common_functions.ping_command(self.floating_ip.ip)
         self.assertTrue(ping, "Instance after creation is not reachable")
+
+
+class TestNovaDeferredDelete(TestBase):
+    """Nova Deferred Delete test cases"""
+
+    @classmethod
+    @pytest.fixture(scope='class', autouse=True)
```

Comment:
After some experiments I see the difference:
WITHOUT @pytest.mark.undestructive:
```
$ py.test --pdb -s -v -ra --snapshot=ha_deploy_VLAN_CINDER --env=akor_Test_Deployment_MOS_CI mos_tests/nova/nova_test.py::TestNovaDeferredDelete
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_1
    2016-04-22 13:53:19,166 [INFO] mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    2016-04-22 13:55:11,131 [DEBUG] mos_tests.nova.nova_test:616: ~~ SETUP ~~
    2016-04-22 13:56:14,997 [DEBUG] mos_tests.nova.nova_test:625: test_1
    PASSED
    2016-04-22 13:56:15,000 [INFO] mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_2
    2016-04-22 13:58:06,983 [DEBUG] mos_tests.nova.nova_test:627: test_2
    PASSED
    2016-04-22 13:58:06,985 [INFO] mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_3
    2016-04-22 13:59:57,950 [DEBUG] mos_tests.nova.nova_test:629: test_3
    PASSED
    2016-04-22 13:59:57,951 [DEBUG] mos_tests.nova.nova_test:622: ~~ TEARDOWN ~~
```
WITH @pytest.mark.undestructive:
```
$ py.test --pdb -s -v -ra --snapshot=ha_deploy_VLAN_CINDER --env=akor_Test_Deployment_MOS_CI mos_tests/nova/nova_test.py::TestNovaDeferredDelete
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_1 
    2016-04-22 16:22:10,727 [INFO] mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    22016-04-22 16:24:36,068 [DEBUG] mos_tests.nova.nova_test:617: ~~ SETUP ~~
    2016-04-22 16:25:38,994 [DEBUG] mos_tests.nova.nova_test:626: test_1
    PASSED
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_2 
    2016-04-22 16:25:41,458 [DEBUG] mos_tests.nova.nova_test:628: test_2
    PASSED
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_3 
    2016-04-22 16:25:43,800 [DEBUG] mos_tests.nova.nova_test:630: test_3
    PASSED
    2016-04-22 16:25:43,801 [DEBUG] mos_tests.nova.nova_test:623: ~~ TEARDOWN ~~
```